source/FreeRTOS_DNS_Parser.c

```diff
@@ -71,13 +71,11 @@
                 else
             #endif /* ( ipconfigUSE_IPv6 != 0 ) */
 
-            #if ( ipconfigUSE_IPv4 != 0 )
-                {
-                    IPPacket_t * xIPPacket = ( ( IPPacket_t * ) pxNetworkBuffer->pucEthernetBuffer );
+            {
```

Comment:
`ipconfigUSE_IPv4` should be always enabled when DNS is used, hence this check doesn't matter.